File: shared/src/preAction.ts
import type { ActionType } from './types/hand.js';

export type PreActionType = 'fold_to_bet' | 'auto_check';

export function resolvePreAction(
  preAction: PreActionType | null,
  availableActions: ActionType[],
): ActionType | null {
  if (!preAction) return null;

  if (preAction === 'fold_to_bet') {
    return !availableActions.includes('check') ? 'fold' : null;
  }

  if (preAction === 'auto_check') {
    return availableActions.includes('check') ? 'check' : null;
  }

  return null;
}

File: ios/PokerSofta/Sources/Utilities/PreAction.swift
import Foundation

enum PreActionType {
    case checkFold
    case autoCheck
}

func resolvePreAction(_ preAction: PreActionType?, available: [ActionType]) -> ActionType? {
    guard let preAction else { return nil }

    switch preAction {
    case .checkFold:
        return available.contains(.check) ? .check : .fold
    case .autoCheck:
        return available.contains(.check) ? .check : nil
    }
}
